src/DataViews/LiveFunctionsDataViewTest.cpp

```diff
@@ -0,0 +1,756 @@
+  }
+
+  // Disable frame track(s)
+  {
+    const auto disable_frame_track_index =
+        std::find(context_menu.begin(), context_menu.end(), "Disable frame track(s)") -
+        context_menu.begin();
+    ASSERT_LT(disable_frame_track_index, context_menu.size());
+
+    EXPECT_CALL(app_, DisableFrameTrack).Times(1).WillOnce([&](const FunctionInfo& function) {
+      EXPECT_EQ(function.name(), kNames[0]);
+    });
+    EXPECT_CALL(app_, RemoveFrameTrack(testing::An<uint64_t>()))
+        .Times(1)
+        .WillOnce([&](uint64_t function_id) { EXPECT_EQ(function_id, kFunctionIds[0]); });
+    view_.OnContextMenu("Disable frame track(s)", static_cast<int>(disable_frame_track_index), {0});
+  }
+}
+
+TEST_F(LiveFunctionsDataViewTest, FilteringShowsRightResults) {
+  AddFunctionsByIndices({0, 1, 2});
+  EXPECT_CALL(app_, HasCaptureData).WillRepeatedly(testing::Return(true));
+  EXPECT_CALL(app_, GetCaptureData).WillRepeatedly(testing::ReturnRef(*capture_data_));
+
+  // Filtering by function display name with single token
+  {
+    EXPECT_CALL(app_, SetVisibleFunctionIds)
+        .Times(1)
+        .WillOnce([&](absl::flat_hash_set<uint64_t> visible_function_ids) {
+          EXPECT_THAT(visible_function_ids,
+                      testing::UnorderedElementsAre(kFunctionIds[1], kFunctionIds[2]));
+        });
+    view_.OnFilter("int");
+    EXPECT_EQ(view_.GetNumElements(), 2);
+    EXPECT_THAT((std::array{view_.GetValue(0, kColumnName), view_.GetValue(1, kColumnName)}),
+                testing::UnorderedElementsAre(kPrettyNames[1], kPrettyNames[2]));
+  }
+
+  // Filtering by function display name with multiple tokens separated by " "
+  {
+    EXPECT_CALL(app_, SetVisibleFunctionIds)
+        .Times(1)
+        .WillOnce([&](absl::flat_hash_set<uint64_t> visible_function_ids) {
+          EXPECT_THAT(visible_function_ids, testing::UnorderedElementsAre(kFunctionIds[1]));
+        });
+
+    view_.OnFilter("int main");
+    EXPECT_EQ(view_.GetNumElements(), 1);
+    EXPECT_EQ(view_.GetValue(0, kColumnName), kPrettyNames[1]);
+  }
+
+  // No matching result
+  {
+    EXPECT_CALL(app_, SetVisibleFunctionIds)
+        .Times(1)
+        .WillOnce([](absl::flat_hash_set<uint64_t> visible_function_ids) {
+          EXPECT_TRUE(visible_function_ids.empty());
+        });
+    view_.OnFilter("int module");
+    EXPECT_EQ(view_.GetNumElements(), 0);
+  }
+}
+
+TEST_F(LiveFunctionsDataViewTest, UpdateHighlightedFunctionsOnSelect) {
+  AddFunctionsByIndices({0, 1, 2});
+
+  EXPECT_CALL(app_, DeselectTimer).Times(3);
+  EXPECT_CALL(app_, GetHighlightedFunctionId).Times(3);
+  EXPECT_CALL(app_, HasCaptureData).WillRepeatedly(testing::Return(true));
+
+  // Single selection
+  {
+    EXPECT_CALL(app_, SetHighlightedFunctionId)
+        .Times(1)
+        .WillOnce([&](uint64_t highlighted_function_id) {
+          EXPECT_EQ(highlighted_function_id, kFunctionIds[2]);
+        });
+
+    view_.OnSelect({2});
+  }
+
+  // Multiple selection
```

Comment:
Can you document the expected behaviour here? I assume we expect that the first selected function will be set as the highlighted function?

Reply:
I updated the comments here. But I notice that for the current selection in views, we use `std::set` to remove the duplicate selected indices, but `std::std` will also sort the selected indices. This means that the `LiveFunctionsDataView::OnSelect` always receive sorted rows. I will fix this in another PR.